File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        len_s = len(s)
        len_t = len(t)
        if len_s != len_t:
            return False

        d1, d2 = {}, {}

        for ch1, ch2 in zip(s, t):
            d1[ch1] = d1.get(ch1, 0) + 1
            d2[ch2] = d2.get(ch2, 0) + 1
        
        if d1 == d2:
            return True
        return False
        
        # if s.sort() == t.sort():
        #     return True
        # return False

File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        int freq[26] = {};

        if(s.size() != t.size()) return false;

        for(int i = 0; i < s.size(); i++) {
            freq[s[i] - 'a'] += 1;
            freq[t[i] - 'a'] -= 1;
        }

        for(auto item : freq) {
            if(item != 0) return false;
        }

        return true;
    }
};